x-pack/plugins/security_solution/common/endpoint/data_loaders/index_endpoint_hosts.ts

```diff
@@ -0,0 +1,312 @@
+      if (!realPolicies[appliedPolicyId]) {
+        const createdPolicies = await indexFleetEndpointPolicy(
+          kbnClient,
+          appliedPolicyName,
+          epmEndpointPackage.version
+        );
+
+        merge(response, createdPolicies);
+
+        // eslint-disable-next-line require-atomic-updates
+        realPolicies[appliedPolicyId] = createdPolicies.integrationPolicies[0];
+      }
+
+      // If we did not yet enroll an agent for this Host, do it now that we have good policy id
+      if (!wasAgentEnrolled) {
+        wasAgentEnrolled = true;
+
+        const indexedAgentResponse = await indexFleetAgentForHost(
+          client,
+          kbnClient,
+          hostMetadata!,
+          realPolicies[appliedPolicyId].policy_id,
+          kibanaVersion
+        );
+
+        enrolledAgent = indexedAgentResponse.agents[0];
+        merge(response, indexedAgentResponse);
+      }
+      // Update the Host metadata record with the ID of the "real" policy along with the enrolled agent id
+      hostMetadata = {
+        ...hostMetadata,
+        elastic: {
+          ...hostMetadata.elastic,
+          agent: {
+            ...hostMetadata.elastic.agent,
+            id: enrolledAgent?.id ?? hostMetadata.elastic.agent.id,
+          },
+        },
+        Endpoint: {
+          ...hostMetadata.Endpoint,
+          policy: {
+            ...hostMetadata.Endpoint.policy,
+            applied: {
+              ...hostMetadata.Endpoint.policy.applied,
+              id: realPolicies[appliedPolicyId].id,
+            },
+          },
+        },
+      };
+
+      // Create some actions for this Host
+      await indexFleetActionsForHost(client, hostMetadata);
+    }
+
+    await client
+      .index({
+        index: metadataIndex,
+        body: hostMetadata,
+        op_type: 'create',
+      })
+      .catch(wrapErrorAndRejectPromise);
+
+    const hostPolicyResponse = generator.generatePolicyResponse({
+      ts: timestamp - timeBetweenDocs * (numDocs - j - 1),
+      policyDataStream: EndpointDocGenerator.createDataStreamFromIndex(policyResponseIndex),
+    });
+
+    await client
+      .index({
+        index: policyResponseIndex,
+        body: hostPolicyResponse,
+        op_type: 'create',
+      })
+      .catch(wrapErrorAndRejectPromise);
+
+    // Clone the hostMetadata and policyResponse document to ensure that no shared state
+    // (as a result of using the generator) is returned across docs.
+    response.hosts.push(cloneDeep(hostMetadata));
+    response.policyResponses.push(cloneDeep(hostPolicyResponse));
+  }
+
+  return response;
+}
+
+const fetchKibanaVersion = async (kbnClient: KbnClient) => {
+  const version = ((await kbnClient.request({
+    path: '/api/status',
+    method: 'GET',
+  })) as AxiosResponse).data.version.number;
+
+  if (!version) {
+    // eslint-disable-next-line no-console
+    console.log('failed to retrieve kibana version');
+    return '8.0.0';
```

Comment:
any chance to read this from the package.json?

Reply:
I guess I could, but not sure that I should. Because we can actually run these set of utility agains any environment (not only local), we really do need the version of Kibana from the system we are interacting with. I'm actually thinking that this here should actually `throw` instead of defaulting to 8.0.0.

Reply:
throwing then indeed sounds like a better idea than the hard-coded version